Makefile

```diff
@@ -192,10 +192,15 @@ generate:
 	go generate -x ./...
 	$(MAKE) format
 
-# Generate descriptors for supported ENI configurations.
+# Generate limit file go code
 generate-limits: GOOS=
 generate-limits:
-	go run pkg/awsutils/gen_vpc_ip_limits.go
+	go run scripts/gen_vpc_ip_limits.go
+
+# Generate eni-max-pods.txt file for EKS AMI
+generate-max-pods: GOOS=
+generate-max-pods: 
+	GENERATE_MAX_PODS=true go run scripts/gen_vpc_ip_limits.go
```

Comment:
Too fancy.  Just (re)generate both output files always.

Reply:
Hmm... The issue here is that the `eni-max-pods.txt` file actually [belongs to the EKS AMI package](https://github.com/awslabs/amazon-eks-ami/blob/master/files/eni-max-pods.txt)... I just added the functionality here to generate it, using the same code that already generated the CNI limits. Since these are flags to the Kubelet on startup, it's not as easy to fetch the limits there.

Reply:
It's just a loop over the results of a single API call - any benefit isn't worth carrying extra unused code.  Either we invoke (and therefore exercise) both code paths all the time, or the code path that this repo doesn't care about should be removed.

I haven't dug deeply enough to understand how this glues into the AMI build process, but afaics it's one of:
- we don't need eni-max-pods.txt from this binary: remove the code branch
- we need to generate eni-max-pods.txt from this binary during AMI build: generate it always, and .gitignore it from this repo (with a comment).
- we need to pull the already-generated eni-max-pods.txt from this repo during AMI build: generate it always and commit it to this repo (with a comment).

Reply:
@anguslees We already have a dependency the other way, where the CNI will [download the debug script from the EKS AMI repo](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/Makefile#L212-L221). I've changed this PR to use `int` and to generate both files, checking in `misc/eni-max-pods.txt`.